tests/test_gso.py

```diff
@@ -109,3 +116,44 @@ def test_gso_io():
                 v_ = IntegerMatrix.from_iterable(1, m, w) * A
                 v_ = list(v_[0])
                 assert v == v_
+
+def test_gso_coherence_gram_matrix():
+    """
+        Test if the GSO is coherent if it is given a matrix A or its associated
+        Gram matrix A*A^T
+    """
+
+    EPSILON = 0.0001
+
+    for m, n in dimensions:
+        for int_type in int_types:
+            # long is not tested for high dimensions because of integer overflow
+            if m > 20 and int_type == "long":
+                continue
+
+            A = make_integer_matrix(m, n, int_type=int_type).transpose()
+            G = tools.compute_gram(A)
+
+            for float_type in float_types:
+                M_A = GSO.Mat(copy(A), float_type=float_type, gram=False, flags=GSO.INT_GRAM)
+                M_A.update_gso()
+
+                M_G = GSO.Mat(copy(G), float_type=float_type, gram=True, flags=GSO.INT_GRAM)
+                M_G.update_gso()
+
+                # Check that the gram matrix coincide
+                for i in range(m):
+                    for j in range(i):
+                        assert M_A.get_int_gram(i, j) == G[i, j]
+
+                # Check if computations coincide
+                for i in range(m):
+                    for j in range(i):
+                        assert (
+                            M_A.get_r(i, j) == 
+                            pytest.approx(M_G.get_r(i, j), rel=EPSILON)
```

Comment:
The diagonal get_r(i,i) should also be checked (only for get_r, not for get_mu).

Reply:
Other than that this PR can be merged.

Reply:
Fixed in b0d4ca43b8e66e7fcc3f63519c85b1b261f4defc